File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement = 0
        seen = {}
        for i in range(len(nums)):
            if nums[i] in seen:
                return [seen[nums[i]], i]
            complement = target - nums[i]
            seen[complement] = i

File: Data Structures & Algorithms/largest-unique-number/submission-2.cpp
#include <unordered_map>

class Solution {
public:
    int largestUniqueNumber(vector<int>& nums) {
        unordered_map<int, int> seen;

        for(int num : nums) {
            seen[num]++;
        }
        int largest = -1;

        for(auto& entry : seen) {
            if(entry.second == 1) {
                largest = max(largest, entry.first);
            }
        }
        return largest;
    }
};
